File: 724-find-pivot-index/find-pivot-index.js
var pivotIndex = function(nums) {
    var sum = 0;
    for(num of nums) sum+=num

    var left = 0;
    for(let i = 0;i<nums.length;i++){
        if(left == sum-left-nums[i]) return i;
        left+=nums[i];
    }

    return -1;
};

File: 724-find-pivot-index/find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int total = 0;
        for(int i:nums) total+=i;
        int leftTotal = 0;
        for(int i=0;i<nums.length;i++){
            int rightTotal = total-nums[i]-leftTotal;
            if(leftTotal == rightTotal) return i;
            leftTotal+=nums[i];
        }

        return -1;
    }
}